src/runner.ts

```diff
@@ -261,7 +331,10 @@ async function doLinting(options: Options, files: string[], program: ts.Program
         }
 
         let contents: string | undefined;
-        if (program !== undefined) {
+
+        if (resolvedFile.isStdin) {
+            contents = await readStdin();
```

Comment:
is `--std-filename` being ignored when actually fetching content with `--std`?

Reply:
Essentially yes. The filename is a hint for where the file is, and the data comes from stdin.